alpha/patterns/declarative/options.go

```diff
@@ -0,0 +1,151 @@
+package declarative
```

Comment:
We should add pkg level `doc.go` explaining this pattern and the types defined below.